src/EnergyPlus/SetPointManager.cc

```diff
@@ -832,64 +832,76 @@ void GetSetPointManagerInputData(EnergyPlusData &state, bool &ErrorsFound)
-        state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutHigh1 = rNumericArgs(4);
-        state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).CtrlNodeListName = cAlphaArgs(3);
-        if (state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutHigh1 < state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutLow1) {
+        thisOASetPtMgr.OutLowSetPt1 = rNumericArgs(1);
+        thisOASetPtMgr.OutLow1 = rNumericArgs(2);
+        thisOASetPtMgr.OutHighSetPt1 = rNumericArgs(3);
+        thisOASetPtMgr.OutHigh1 = rNumericArgs(4);
+        thisOASetPtMgr.CtrlNodeListName = cAlphaArgs(3);
+        if (thisOASetPtMgr.OutHigh1 < thisOASetPtMgr.OutLow1) {
             ShowWarningError(state, format("{}: {}=\"{}\", invalid field.", RoutineName, cCurrentModuleObject, cAlphaArgs(1)));
             ShowContinueError(state,
                               format("...{}=[{:.1R}] is less than {}=[{:.1R}].",
                                      cNumericFieldNames(4),
-                                     state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutHigh1,
+                                     thisOASetPtMgr.OutHigh1,
                                      cNumericFieldNames(2),
-                                     state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutLow1));
+                                     thisOASetPtMgr.OutLow1));
         }
         // Get optional input: schedule and 2nd reset rule
         if (NumAlphas == 4 && NumNums == 8) {
-            state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).Sched = cAlphaArgs(4);
-            state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).SchedPtr = GetScheduleIndex(state, cAlphaArgs(4));
-            // Schedule is optional here, so no check on SchedPtr
-            state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutLowSetPt2 = rNumericArgs(5);
-            state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutLow2 = rNumericArgs(6);
-            state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutHighSetPt2 = rNumericArgs(7);
-            state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutHigh2 = rNumericArgs(8);
-            if (state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutHigh2 < state.dataSetPointManager->OutAirSetPtMgr(SetPtMgrNum).OutLow2) {
+            thisOASetPtMgr.SchedPtr = GetScheduleIndex(state, cAlphaArgs(4)); // Schedule is optional here, so no check on SchedPtr
+            if (thisOASetPtMgr.SchedPtr > 0) {                                // Only check schedule values if a schedule was entered
+                Real64 minValSched = GetScheduleMinValue(state, thisOASetPtMgr.SchedPtr);
+                Real64 maxValSched = GetScheduleMaxValue(state, thisOASetPtMgr.SchedPtr);
+                if ((minValSched < 1.0) || (maxValSched > 2.0)) {
+                    ShowSevereError(state, format("{}: {}=\"{}\", invalid field.", RoutineName, cCurrentModuleObject, cAlphaArgs(1)));
```

Comment:
Reminder for the future. Change cAlphaArgs(1) to thisOASetPtMgr.Name. All getObjectItem data is replace with struct var after first occurance.